src/Workspaces/Core/Portable/SymbolKey/SymbolKey.cs

```diff
@@ -100,33 +98,13 @@ public static IEqualityComparer<SymbolKey> GetComparer(bool ignoreCase, bool ign
             return reader.RemoveAssemblySymbolKeys();
         };
 
-        /// <summary>
-        /// Tries to resolve the provided <paramref name="symbolKey"/> in the given 
-        /// <paramref name="compilation"/> to a matching symbol.  <paramref name="resolveLocations"/>
-        /// should only be given <see langword="true"/> if the symbol was produced from a compilation
-        /// that has the exact same source as the compilation we're resolving against.  Otherwise
-        /// the locations resolved may not actually be correct in the final compilation.
-        /// </summary>
-        public static SymbolKeyResolution Resolve(
-            string symbolKey, Compilation compilation,
-            bool ignoreAssemblyKey = false, bool resolveLocations = false,
-            CancellationToken cancellationToken = default)
-        {
-            using (var reader = SymbolKeyReader.GetReader(
-                symbolKey, compilation, ignoreAssemblyKey, resolveLocations, cancellationToken))
-            {
-                var result = reader.ReadFirstSymbolKey();
-                Debug.Assert(reader.Position == symbolKey.Length);
-                return result;
-            }
-        }
+        public static SymbolKey From(ISymbol symbol, CancellationToken cancellationToken = default)
+            => new SymbolKey(Encode(symbol, cancellationToken));
 
-        public static SymbolKey Create(ISymbol symbol, CancellationToken cancellationToken = default)
-        {
-            return new SymbolKey(ToString(symbol, cancellationToken));
-        }
+        public static SymbolKey From(string encodedSymbolData)
+            => new SymbolKey(encodedSymbolData);
 
-        public static string ToString(ISymbol symbol, CancellationToken cancellationToken = default)
+        public static string Encode(ISymbol symbol, CancellationToken cancellationToken = default)
```

Comment:
if we choose the name 'Encode', then i think 'Resolve' should be named 'Decode'.

Reply:
`Resolve` doesn't take the output of `Encode`, so `Decode` really isn't appropriate for that.

Reply:
Maybe changing this name to `GetEncodedSymbolData` would alleviate the need to have a `Decode` method.

Reply:
> We could consider just having this all be strings. with statics to Encode/Decode those strings...

That's possible. In that case, we could provide SymbolEncoder and SymbolDecoder APIs. However, I dislike the `Decode` because it's not guaranteed to work since it's "decoding" in a different context. When I see `Encode`/`Decode`, I expect them to always work.

Reply:
IMO, `Resolve` is better at setting expectations.

Reply:
One of the reason's I'm ok with Decode (or maybe TryDecode) is that it returns a SymbolInfo, which already sets up the expectation that it might not work.  But your point is well taken and it warms me a bit to Encode/Resolve.

Reply:
> We can't hide that this is just a string (after all, we expose the string otday, and allow you to construct SymbolKey with it). So i think we should just embrace this as a supported way to go to/from a string, and model the API around that.

My head is in that same space. Exposing the strings is critical to meet the scenario, unless we want to make this struct serializable. Although, I'm still thinking that there may still be value in the struct to help people use the API successfully.

If only you still worked here, we could chat about this in person.

Reply:
:D

I'm happy with this online discussion.  Sounds like you're considering all the same things that concern me.  I think we'll be able to come up with an excellent solution here that covers all the areas we've identified.

Plus, now everyone can be part of the convo :D